presto-main/src/main/java/com/facebook/presto/cost/UnionStatsRule.java

```diff
@@ -55,7 +56,7 @@ public UnionStatsRule(StatsNormalizer normalizer)
             PlanNode source = node.getSources().get(i);
             PlanNodeStatsEstimate sourceStats = statsProvider.getStats(source);
 
-            PlanNodeStatsEstimate sourceStatsWithMappedSymbols = mapToOutputSymbols(sourceStats, node.getVariableMapping(), i);
+            PlanNodeStatsEstimate sourceStatsWithMappedSymbols = mapToOutputSymbols(sourceStats, toListMultimap(node.getVariableMapping()), i);
```

Comment:
No need to `toListMultimap` to make another copy, we can just update `mapToOutputSymbols` to use `Map<VariableReferenceExpression, List<VariableReferenceExpression>>`. Same for many other places use `toListMultimap`.